Update instructions for updating dev environment. (#7160)
scikit-image/scikit-image: INSTALL.rst

Updating the installation
------------------------------------------------------------------------------

Before updating your installation, you typically want to grab the latest
source:

.. code-block:: sh

   git checkout main
   git pull upstream main

> And you possibly want to create a feature branch from there (or check out an
> existing branch where you are testing changes). As you work on this branch,
> you can re-build scikit-image with the following command:

Reviewer: ```suggestion
And you likely want to create a feature branch from there.
As you work on this branch, you can re-build scikit-image using:
```

.. code-block:: sh

   spin build

> If you need to clean the build directory before building scikit-image (e.g.,

Reviewer: How about something:

"Mostly, repeated, incremental builds work just fine, but if you notice strange build problems,
rebuild from scratch using:"

Author: Thanks, @stefanv. Editing your suggestion slightly, how about:

"Repeated, incremental builds usually work just fine, but if you notice build problems,
rebuild from scratch using:"

(I find that the word 'strange' creates some unnecessary mystification...)

Reviewer: :ghost:
